### DISCUSSION TITLE
Resetting fields in form

### DISCUSSION THREAD

**Discussion by titantwentyone** [livewire/livewire#7560] [1. Help]
[Accotding to the docs](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#resetting-a-forms-data), we can reset a form with just `$this->form->fill()`. The component below is using this but the fields with the form are not being reset. Unsure if this is a bug/documentation issue/my code.

```php
class TestComponent extends Component implements HasForms
{
    use InteractsWithForms;

    public ?array $data;
    public string $message = '';

    public function mount()
    {
        $this->form->fill();
    }

    public function form(Form $form): Form
    {
        return $form
            ->statePath('data')
            ->schema([
                TextInput::make('testing')
                    ->live()
            ]);
    }

    public function display()
    {
        $this->message = $this->data['testing'];
        $this->form->fill();
    }

    public function render()
    {
        return view('my.test-component');
    }
}
```
```blade
<div>
    {{ $this->form }}
    <button type='button' wire:click='display'>Reset</button>
    {{ $this->message }}
</div>
```
There is a similar issue [here](https://github.com/filamentphp/filament/discussions/8979) but the author did not elaborate. Anyone else come across this? Any temp workarounds? Any help appreciated!

**Comment by titantwentyone** [comment]
A thousand apologies. Wrong repo, Been a long day...!

**Reply by joshhanley** [reply]
No worries! 🙂

**Comment by hamannjames** [comment]
There needs to be better documentation about this in the livewire docs. I spent about an hour trying to figure out a good way to restore the default state of a form without emptying the values or doing some hybrid alpine thing. I didn't see anywhere in the official v3 docs that mentioned restoring to initial state using fill(). Is that not an intended feature @calebporzio? Or maybe I just missed it.

**Reply by hamannjames** [reply]
If it needs to be added to the docs, I'd be happy to do it via a PR

**Reply by joshhanley** [reply]
@hamannjames That's not intended. I would use reset instead. `$this->form->reset();`. 

Re the docs, all PRs are welcome if you think something isn't clear or needs to be added 🙂

**Reply by hamannjames** [reply]
@joshhanley I was having trouble with resetting the form to an initial state if it was an "update" form, like with the example in the docs where you pass an eloquent model to the form to save to. It was nullifying the model, and resetting the form fields to empty values. This made me reach for fill.

I dug a bit deeper after your comment though and realized that I could call reset and then call the setter methods to set the eloquent model on the form again, and that works. You can do that externally or but overriding the form reset method with your own that calls parent then does the setting.

I might do a PR with the docs just to make this more clear. It seems obvious now, but wasn't obvious up front

**Reply by joshhanley** [reply]
@hamannjames ah right, yep that works, there is also a `resetExcept()` method where you could specify the model property to ensure it doesn't get reset.
